srv/salt/_modules/osd.py

```diff
@@ -797,6 +797,12 @@ def clean(self):
 
         Note: expected to only run inside of "not is_prepared"
         """
+        if (self.osd.disk_format != 'filestore' and
+            self.osd.disk_format != 'bluestore'):
```

Comment:
In which circumstances did you see that happening?

Reply:
I did this primarily for the smoketests.  Overriding Salt dictionaries had some issues, so I set a disk_format of 'none'.  

On the chance that an admin does mistype a format, it's better not to wipe all the partitions on that drive until it's corrected.